hack/externalnode/install-vm.sh

```diff
@@ -141,17 +153,280 @@ EOF
+    docker version
+    if [[ "$?" -ne 0 ]]; then
+        install_container_runtime
+    fi
+    # Install docker-compose
+    docker-compose version
+    if [ "$?" -ne 0 ]; then
+        cmd="curl -L "${DOCKER_COMPOSE}-$(uname -s)-$(uname -m)" -o ${DOCKER_COMPOSE_DIR}"
+        execute_command "$cmd" 1m
+        if [ "$?" -eq 0 ]; then
+            echo "docker-compose is downloaded successfully"
+            chmod +x ${DOCKER_COMPOSE_DIR}
+            ln -s ${DOCKER_COMPOSE_DIR} /usr/bin/docker-compose
+        else
+            echoerr "Error docker-compose couldn't be downloaded, exiting"
+            exit 2
+        fi
+    fi
+    set -eo pipefail
+}
+
+# Create docker-compose file with OVS and antrea-agent container specific parameters.
+create_docker_compose_file() {
+    cat << EOF > $AGENT_CONF_PATH/docker-compose.yaml
+version: "3"
+
+services:
+  ovs:
+    image: $ANTREA_DOCKER_IMAGE
+    container_name: ovs
+    volumes:
+        - /var/log/openvswitch:/var/log/openvswitch
+        - /var/run/openvswitch:/var/run/openvswitch
+        - /lib/modules:/lib/modules
+    command: $OVS_BIN
+    network_mode: host
+    cap_add:
+        - NET_ADMIN
+        - SYS_NICE
+        - SYS_ADMIN
+        - IPC_LOCK
+
+  antrea-agent:
+    image: $ANTREA_DOCKER_IMAGE
+    container_name: antrea-agent
+    volumes:
+        - /etc/antrea/:/etc/antrea
+        - /var/run/antrea:/var/run/antrea
+        - /var/run/openvswitch:/var/run/openvswitch
+        - /var/log/antrea:/var/log/antrea
+    environment:
+        - NODE_NAME=$NODE_NAME
+    command: $ANTREA_AGENT_BIN --config $AGENT_CONF_PATH/antrea-agent.conf --logtostderr=false --log_file=$AGENT_LOG_DIR/antrea-agent.log
+    network_mode: host
+    privileged: true
+    cap_add:
+        - SYS_MODULE
+EOF
+}
+
+create_antrea_agent_service_file() {
+    cat >$AGENT_CONF_PATH/antrea-agent.sh << 'EOF'
+#!/bin/bash
+
+AGENT_CONF_PATH="/etc/antrea"
+AGENT_LOG_DIR="/var/log/antrea"
+PIDFILE="$AGENT_CONF_PATH/antrea-agent.pid"
+DOCKER_COMPOSE_YAML="$AGENT_CONF_PATH/docker-compose.yaml"
+COUNT=0
+LIMIT=1000
+
+declare -a OVS_KERNEL_MODULES=("openvswitch")
+declare -a CONTAINERS=("ovs" "antrea-agent")
+
+remove_pidfile() {
+    rm -f "$PIDFILE"
+}
+
+load_ovs_modules() {
+    for module in "${OVS_KERNEL_MODULES[@]}"; do
+        lsmod | grep -q $module
+        if [[ $? -ne 0 ]]; then
+            echo "Load $module kernel module"
+            modprobe $module
+            if [ "$?" -ne 0 ]; then
+                echoerr "Error loading kernel module $module"
+            fi
+        fi
+    done
+}
+
+start_containers() {
+    while [ $COUNT -lt $LIMIT ] ; do
+        for service in "${CONTAINERS[@]}"; do
+            # Check if containers are UP
+            if [ ! "$(docker ps -q -f name=$service)" ]; then
+                if [ "$(docker ps -aq -f status=exited -f name=$service)" ] ; then
+                    echo "$service container is in exited state"
```

Comment:
why it restarts the container manually, instead of configuring the restart policy?

Reply:
Yeah, we can use that. We also thought of the same approach but for some reason, we implemented this. We implemented, like you suggested.